src/CustomerioTracking.tsx

```diff
@@ -143,6 +143,39 @@ class CustomerIO {
     }
     CustomerioReactnative.registerDeviceToken(token)
   }
+
+  /**
+   * Request to show prompt for push notification permissions.
+   * Prompt will only be shown if the current status is - not determined.
+   * In other cases, this function will return current status of permission.
+   * @param options
+   * @returns Success & Failure promises
+   */
+  static async showPromptForPushNotifications(options?: any) : Promise<any>{
+    let pushConfigurationOptions  = {
+      "ios": {
+        "badge" : true,
+        "sound" : true
+      }
+    }
+    if (typeof options !== 'undefined'){
+      pushConfigurationOptions = options
+    }
+  
+  return CustomerioReactnative.showPromptForPushNotifications(pushConfigurationOptions)
+  }
+
+  /**
+   * Get status of push permission for the app
+   * @param handler
+   * @returns Status of push permission as a string
+   */
+  static getPushPermissionStatus(handler? : Function) {
```

Comment:
Can this function also return a `Promise`? Is there a reason why having a completion handler function would be better then a promise here?

Reply:
With a `Promise` we need a resolver and rejector which in this case might not be relevant. Fetching status will just return the authorization state and nothing else which can be handled with a callback.

Reply:
I understand that this function may not need to reject anything. I do still believe a `Promise` would be valuable. When I write TS code, I appreciate Promise APIs to avoid callback hell. Having some async functions in our SDK use a Promise and some async functions not use a Promise seems inconsistent to me. I think it would be ideal to have all async functions use a Promise based API.

Reply:
Changed from `Callback` to `Promise` for sake of consistency with just `resolver` and no `rejecter`. Changes have been made across all platforms